xenmgr/XenMgr/Connect/Xl.hs

```diff
@@ -188,10 +199,14 @@ start uuid =
       state <- state uuid
       case state of
         Shutdown -> do
-                      exitCode <- system ("xl create " ++ configPath uuid ++ " -p")
-                      case exitCode of
+                      (_, _, Just err, handle) <- createProcess (proc "xl" ["create", configPath uuid, "-p"]){std_err = CreatePipe}
+                      ec <- waitForProcess handle
+                      stderr <- hGetContents err
+                      case ec of
                         ExitSuccess -> return ()
-                        _           -> error "error creating domain"
+                        ExitFailure rc ->
```

Comment:
I'll be honest I don't remember writing that.  Thanks for the heads up, that ExitFailure case can be just _ -> throw ... instead